packages/core/src/html-placeholder.ts

```diff
@@ -70,12 +74,20 @@ export default class Placeholder {
     let newTag = this.end.ownerDocument.createElement('link');
     newTag.href = href;
     newTag.rel = 'stylesheet';
-    this.insert(newTag);
-    this.insertNewline();
+
+    if (this.isScript()) {
+      // Add dynamic styles from scripts to the bottom of the head, and not to where the script was,
+      // to prevent FOUC when pre-rendering (FastBoot)
+      this.appendToHead(newTag);
+    } else {
+      // Keep the new style in the same place as the original one
+      this.insert(newTag);
```

Comment:
This should ensure that styles already present in index.html will be kept where they are, to keep the existing order. Especially when dealing with external styles, that are not funneled through this. (e.g. vendor.css, then external  google-font-css, then app.css). All other dynamically added styles will be appended to `<head>`, in whatever order webpack gives us. 

This is all we can do here about ensuring the "right" order, is it?